Userland/Libraries/LibGUI/EditingEngine.cpp

```diff
@@ -616,7 +616,9 @@ void EditingEngine::move_to_beginning_of_previous_word()
 
             if (column_index == 0 && !is_first_iteration && (vim_isalnum(current_char) || vim_ispunct(current_char)))
                 return m_editor->set_cursor({ line_index, column_index });
-            else if (column_index == 0) {
+            else if (line_index == 0 && column_index == 0) {
+                return m_editor->set_cursor({ line_index, column_index });
+            } else if (column_index == 0 && is_first_iteration) {
                 is_first_iteration = false;
                 continue;
             }
```

Comment:
Please don't mix braced and braceless in the same if/else block. If any of them have braces, they should all have braces :)